tests/pc/test_lag_2.py

```diff
@@ -439,13 +439,16 @@ def test_lag_db_status(duthosts, enum_dut_portchannel_with_completeness_level, i
                         # Sometimes, it has to wait seconds for booting up interface
                         pytest_assert(wait_until(10, 1, 0, check_link_is_up, duthost, po_intf, port_info, lag_name),
                             "{} member {}'s status or netdev_oper_status in state_db is not up.".format(lag_name, po_intf))
-            else:
-                pytest.fail("No valid dut selected, duthost = {}".format(dut_name))
     finally:
         # Recover interfaces in case of failure
-        for lag_name in test_lags:
-            for po_intf in lag_facts['lags'][lag_name]['po_stats']['ports'].keys():
-                duthost.no_shutdown(po_intf)
+        for duthost in duthosts:
```

Comment:
@ZhaohuiS could we have a common function which can decode and return the duthost and dut_lag instead of dut_name, dut_lag? By which we can hide the for loops details to reduce one indent and reuse the code.

Reply:
@StormLiangMS Good catch, I improve the code to reduce one indent and don't have to loop duthost twice. Please review it again, thanks.

Reply:
@StormLiangMS Add one common function `get_duthost_with_name `to return duthost, reduce two indents. Please review.